Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class GameManager : MonoBehaviour {
+
+    [SerializeField]
+    int playerLives = 2;
+
+    [SerializeField]
+    float endDelay = 0f;
+
+    [SerializeField]
+    GameObject playerPrefab;
+
+    private GameObject player;
+    private GameObject[] spawnLocations;
+    private bool gameOver;
+    private WaitForSeconds endWait;
+
+    void Start () {
+        spawnLocations = GameObject.FindGameObjectsWithTag("PlayerSpawn");
+        gameOver = false;
+        endWait = new WaitForSeconds(endDelay);
+
+        SpawnPlayer();
+
+        StartCoroutine(GameLoop());
+	}
+
+    public void KillPlayer() {
+        playerLives--;
+        RespawnPlayer();
+    }
+
+    public void RespawnPlayer() {
+        Transform spawnPoint = GetRandomSpawnPoint();
+        //TODO: Jack needs to fix his hacky ass shit in bb8
+        player.transform.GetChild(0).position = spawnPoint.position;
+        player.transform.GetChild(0).rotation = spawnPoint.rotation;
+        Rigidbody playerRB = player.GetComponentInChildren<Rigidbody>();
+        playerRB.velocity = Vector3.zero;
+        playerRB.angularVelocity = Vector3.zero;
+        //END Jacks hacky shit
+    }
+
+    private void SpawnPlayer() {
+        if(spawnLocations.Length > 0) {
+            player = Instantiate(playerPrefab);
+            RespawnPlayer();
+        } else {
+            Debug.LogError("No Spawn Points set");
+        }
+        
+    }
+
+    private Transform GetRandomSpawnPoint() {
+        int randomLocation = Random.Range(0, spawnLocations.Length);
+        return spawnLocations[randomLocation].transform;
+    }
+
+    private IEnumerator GameLoop() {
+        //yield return StartCoroutine(GameStarting());
+        yield return StartCoroutine(GamePlaying());
+        yield return StartCoroutine(GameEnding());
+
+        if(gameOver) {
+            //TODO: Add game over screen
+            Debug.Log("Game Over");
+            SceneManager.LoadScene(SceneManager.GetActiveScene().buildIndex);
+        }
+
+        StartCoroutine(GameLoop());
```

Comment:
This is what the Tanks example had, I believe it just restarts the game if there isn't a winner, but we might not need this since you will always win or lose.

Reply:
So if both tanks destroyed each other at the same time, it would just restart the game instead of declaring a winner.